### PR TITLE
Do not allow negative source/destination amounts

### PR DESCRIPTION
**Pull request by gip**

### CODE REVIEW

File: src/controllers/quote.js [hunk 1]
```diff
@@ -96,12 +96,12 @@ function validateAmounts (sourceAmount, destinationAmount) {
     throw new NoAmountSpecifiedError('Exactly one of source_amount or destination_amount must be specified')
   }
   if (sourceAmount) {
-    if (isNaN(sourceAmount) || Number(sourceAmount) === 0 ||
+    if (isNaN(sourceAmount) || Number(sourceAmount) === 0 || Number(sourceAmount) < 0 ||
```
Comment (clark800) [inline src/controllers/quote.js]: You can combine to `Number(sourceAmount) <= 0`

Comment (gip) [inline src/controllers/quote.js]: Done

File: test/quoteSpec.js [hunk 2]
```diff
@@ -125,6 +125,32 @@ describe('Quotes', function () {
         .end()
     })
 
+    it('should return a 400 if source_amount is negative', function * () {
+      yield this.request()
+        .get('/quote?' +
+          'destination_amount=-1.3&source_ledger=http://eur-ledger.example' +
```
Comment (clark800) [inline test/quoteSpec.js]: should be `source_amount`?

Comment (gip) [inline test/quoteSpec.js]: Fixed

File: test/quoteSpec.js [hunk 3]
```diff
@@ -125,6 +125,32 @@ describe('Quotes', function () {
...
 
+    it('should return a 400 if source_amount is negative', function * () {
+      yield this.request()
+        .get('/quote?' +
+          'source_amount=-1.3&source_ledger=http://eur-ledger.example' +
+          '&destination_ledger=http://usd-ledger.example')
+        .expect(400)
+        .expect(function (res) {
+          expect(res.body.id).to.equal('InvalidAmountSpecifiedError')
+          expect(res.body.message).to.equal('destination_amount must be finite and positive')
```
Comment (clark800) [inline test/quoteSpec.js]: should change here too

Comment (gip) [inline test/quoteSpec.js]: Done

**Comment by clark800** [comment]
LGTM

File: src/controllers/quote.js [hunk 4]
```diff
@@ -96,12 +96,12 @@ function validateAmounts (sourceAmount, destinationAmount) {
...
   }
   if (sourceAmount) {
-    if (isNaN(sourceAmount) || Number(sourceAmount) === 0 ||
+    if (isNaN(sourceAmount) || Number(sourceAmount) <= 0 ||
       Number(sourceAmount) === Number.POSITIVE_INFINITY) {
       throw new InvalidAmountSpecifiedError('source_amount must be finite and positive')
     }
   } else if (destinationAmount) {
-    if (isNaN(destinationAmount) || Number(destinationAmount) === 0 ||
+    if (isNaN(destinationAmount) || Number(destinationAmount) <= 0 ||
```
Comment (emschwartz) [inline src/controllers/quote.js]: This should probably use a bignum library

**Pull request merged**